subcommand/webhook-cert-manager/command_test.go

```diff
@@ -369,19 +369,16 @@ func TestCertWatcher(t *testing.T) {
 	defer stopCommand(t, &cmd, exitCh)
 
 	ctx := context.Background()
-	// This first sleep is required to ensure the command had completed validating that the MWC does exist in the cluster
-	// It ensures the MWC is absent when the server actually attempts to update it with a bundle.
-	time.Sleep(1 * time.Millisecond)
+	// This sleep ensures the webhook has been updated and the reconcile has been successful.
+	time.Sleep(100 * time.Millisecond)
 	err = k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Delete(ctx, webhookName, metav1.DeleteOptions{})
 	require.NoError(t, err)
 
-	// This sleep ensures the cluster attempts to try updating the MWC every second and fails before we re-create it in the cluster.
-	time.Sleep(100 * time.Millisecond)
 	_, err = k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(ctx, webhook, metav1.CreateOptions{})
 	require.NoError(t, err)
 
```

Comment:
You could patch the webhook and set caCert to "" rather than creating/deleting it.